scripts/ci_test_spdx_license_header.sh

```diff
@@ -3,25 +3,50 @@
 cd $(git rev-parse --show-toplevel)
 
 RET_VAL=0
-FILES=$(find . -iwholename "*.rs")
-CHECK_LICENSE="// SPDX-License-Identifier: Apache-2.0"
-
-for FILE in $FILES
-do
-    FIRST_LINE=$(head -n 1 $FILE)
-    SECOND_LINE=$(head -n 2 $FILE | tail -n 1)
-
-    if [[ "$FIRST_LINE" != "$CHECK_LICENSE" ]]
-    then
-        echo "$FILE :: missing license header \"$CHECK_LICENSE\""
-        RET_VAL=1
-    fi
-
-    if [[ "$SECOND_LINE" != "" ]]
-    then
-        echo "$FILE :: missing new line after license header"
-        RET_VAL=1
-    fi
-done
+
+check_license_header() {
+    FILES=$(find . -type f -iwholename "${FILE_SUFFIX}" )
+    CHECK_LICENSE="${COMMENT_SYMBOL} SPDX-License-Identifier: Apache-2.0"
+
+    for FILE in $FILES
+    do
+        FIRST_LINE=$(head -n 1 $FILE)
+        SECOND_LINE=$(head -n 2 $FILE | tail -n 1)
+
+        if [[ "$FIRST_LINE" != "$CHECK_LICENSE" ]]
+        then
+            echo "$FILE :: missing license header \"$CHECK_LICENSE\""
+            RET_VAL=1
+        fi
+
+        if [[ "$SECOND_LINE" != "" ]]
+        then
+            echo "$FILE :: missing new line after license header"
+            RET_VAL=1
+        fi
+    done
+}
+
+check_rust() {
+    FILE_SUFFIX="*.rs"
+    COMMENT_SYMBOL="//"
+    check_license_header
+}
+
+check_shell() {
+    FILE_SUFFIX="*.sh"
+    COMMENT_SYMBOL="#"
+    check_license_header
+}
+
+check_toml() {
+    FILE_SUFFIX="*.toml"
```

Comment:
Do we really need a license for the toml files? It is just a configuration file. This makes it cumbersome when someone wants to copy a configuration..

Reply:
@elBoberido I just got inspired by zenoh (they do it), but we do not really require this for toml files. I would keep it in the script for now if you have no objections.

Reply:
I would change it. It makes me unconformable to copy stuff from a file with a license without adding that license notice to my own file and it is kind of weird to add those to configuration files